subsys/bootloader/c_runtime_setup/CMakeLists.txt

```diff
@@ -9,39 +9,26 @@ set(CORTEX_M ${ARM}/cortex_m)
 set(EXTS ${ZEPHYR_BASE}/ext)
 set(HAL ${EXTS}/hal)
 
-add_library(c_runtime_setup STATIC ${ZEPHYR_BASE}/misc/empty_file.c)
-target_link_libraries(c_runtime_setup PRIVATE zephyr_interface)
-# Make sure every generated file is present before trying to build the
-# library.
-target_link_libraries(c_runtime_setup PRIVATE bootloader)
+zephyr_library()
 
-target_sources_ifdef(CONFIG_CPU_CORTEX_M0
-  c_runtime_setup
-  PUBLIC
+zephyr_library_sources_ifdef(
+  CONFIG_CPU_CORTEX_M0
   ${ARM}/irq_relay.S
   )
 
-target_sources(
-  c_runtime_setup
-  PRIVATE
+# TODO Use zephyr build code for this directly?
+zephyr_library_sources(
```

Comment:
Try removing it. See arch/arm/core/cortex_m/CMakeLists.txt

Reply:
@sigvartmh probably knows better, but I would think c_runtime_setup can go away completely.

Reply:
Removing it ends up with __start not being defined. We can change this sometime in the future.

Reply:
Regarding `__start` that's being defined in `reset.S` so then it seems you are missing some source files. The long term goal should be to eliminate this file but as for now I don't see a better way of doing this as last time I checked at lot of the arch files were still coupled quite tightly with the kernel.